sortby.rb

```diff
@@ -0,0 +1,38 @@
+def bubble_sort_items(array)
+  items_unsorted = true
+  items_unsorted, array = sorting(array) while items_unsorted
+  array
+end
+
+def sorting(array)
+  sorted = true
+  (0...(array.count - 1)).each do |index|
+    if array[index] > array[index + 1]
+      sorted = false
+      array[index], array[index + 1] = [array[index + 1], array[index]]
+    end
+  end
+  [!sorted, array]
+end
+
+def bubble_sort_by(array)
+  items_unsorted = true
```

Comment:
This method doesn't take a block as an argument and order the elements using the block. Please make the necessary adjustments to fix it. Right now, when you pass a block, it's returning the array sorted but not with the order of the block passed.